### PR TITLE
Update the rosbag2 python module

### PR DESCRIPTION
**Pull request by mjeronimo**
In the rosbag2.py module, fix a few minor issues and add some support functions to:

* Get the sizeof the rosbag
* Get information about a topic (name and type)
* Convert an entry to a ROS message

NOTE: This PR depends on #70 being integrated first.

Signed-off-by: Michael Jeronimo <michael.jeronimo@openrobotics.org>

### CODE REVIEW

**Review by jacobperron — COMMENTED** [review]
LGTM, I've left a couple minor comments.

File: rqt_bag/src/rqt_bag/bag_timeline.py [hunk 1]
```diff
@@ -737,6 +734,10 @@ def _message_recorded(self, topic, msg, t):
                 except Exception as ex:
                     qWarning('Error calling timeline_changed on %s: %s' % (type(listener), str(ex)))
 
+        # Dynamically resize the timeline, if necessary, to make visible any new messages
+        # that might otherwise have exceeded the bounds of the window
+        self.reset_zoom()
+
```
[the remark this answers is not in the document]
Comment (mjeronimo) [reply rqt_bag/src/rqt_bag/bag_timeline.py:740]: Yes, will do.

File: rqt_bag/src/rqt_bag/rosbag2.py [hunk 2]
```diff
@@ -54,13 +56,23 @@ def __init__(self, bag_info, filename):
...
-        bag_dir = os.path.dirname(filename)
+        self.bag_dir = os.path.dirname(filename)
         database_relative_name = bag_info['relative_file_paths'][0]
-        self.db_name = os.path.join(bag_dir, database_relative_name)
+        self.db_name = os.path.join(self.bag_dir, database_relative_name)
 
         self.start_time = Time(nanoseconds=bag_info['starting_time']['nanoseconds_since_epoch'])
         self.duration = Duration(nanoseconds=bag_info['duration']['nanoseconds'])
 
+    def size(self):
```
[the remark this answers is not in the document]
Comment (mjeronimo) [reply rqt_bag/src/rqt_bag/rosbag2.py:66]: Yes, I agree. I'd add any necessary functionality to the rosbag2_py API and remove this class.

File: rqt_bag/src/rqt_bag/rosbag2.py [hunk 3]
```diff
@@ -54,13 +56,23 @@ def __init__(self, bag_info, filename):
...
+        total_size = 0
+        for dirpath, dirnames, filenames in os.walk(self.bag_dir):
+            for filename in filenames:
+                full_name = os.path.join(dirpath, filename)
+                # skip if it is symbolic link
+                if not os.path.islink(full_name):
+                    total_size += os.path.getsize(full_name)
+        return total_size
```
[the remark this answers is not in the document]
Comment (mjeronimo) [reply rqt_bag/src/rqt_bag/rosbag2.py:74]: Nice. I'll use the suggested version.

**Review by jacobperron — APPROVED** [review]

**Pull request merged**